Terminal.Gui/View/Layout/ViewLayout.cs

```diff
@@ -315,6 +315,29 @@ public Pos Y
 
     // EndLayout
 
+    private bool _excludeFromLayout;
+
+    /// <summary>
+    ///  Gets or sets whether this View will be excluded from layout computations. 
+    /// </summary>
+    /// <remarks>
+    /// If set to <see langword="true"/>, the layout engine will ignore this view. Any subviews of this view will still be laid out.
+    /// Any views that have Pos or Dim objects that reference this view must have <see cref="ExcludeFromLayout"/> set to <see langword="true"/> as well
+    /// </remarks>
+    public bool ExcludeFromLayout
+    {
+        get => _excludeFromLayout;
+        set
+        {
+            if (value == _excludeFromLayout)
+            {
+                return;
+            }
+            _excludeFromLayout = value;
```

Comment:
Good place to raise `PropertyChanged`

Reply:
Also, if we either subscribe to the event internally or use an invoker wrapper, the calls to `SetNeedsLayout` in all these ***could*** move into that and would just be replaced by raising the event, in the property handlers.

Or... Getting fancier...

If there were a non-public event (private protected is probably appropriate) called something like "LayoutInvalidated" or whatever makes sense to you, these calls to `SetNeedsLayout()` could be replaced with raising that event, and use `[CallerMemberName]` and the `sender` to communicate who asked and why, as well as enable components to opt in and out of participating if/as needed.